### ISSUE TITLE
Access denied when importing content of external source for a valid submitter

### ISSUE THREAD

**Issue by paulo-graca** [DSpace/DSpace#9797]
## Describe the bug
When submitting content, a user with submission permission (and not an admin) is unable to use the submission form lookup mechanism, because the following occurs:

```
2024-09-03 16:05:09,165 INFO  1a63e368-4e76-4c4b-90ce-cc95daa380b7 df9818be-299b-4f1b-b3d6-3fc6405c0a68 org.dspace.app.rest.utils.DSpaceAPIRequestLoggingFilter @ Before request [POST /server/api/core/items, client=IP_ADDRESS, user=USER@EMAIL.pt] originated from /workspaceitems/25610/edit?spc.page=2&spc.rpp=5&query=paulo%20gra%C3%A7a
2024-09-03 16:05:09,169 WARN  1a63e368-4e76-4c4b-90ce-cc95daa380b7 df9818be-299b-4f1b-b3d6-3fc6405c0a68 org.dspace.app.rest.exception.DSpaceApiExceptionControllerAdvice @ Access is denied (status:403 exception: org.dspace.authorize.AuthorizeException: Only admins are allowed to create items using external data at: org.dspace.app.rest.repository.DSpaceRestRepository.createAndReturn(DSpaceRestRepository.java:351))
```

## To Reproduce
1. Using DSpace 7.6.1
2. Your user can't be an admin, but needs to have submission permissions for a particular collection
3. On the front-end, try to deposit an entity of type Publication on that Collection
4. For that Publication, use the front end's lookup mechanism and try to fetch an Orcid record to your repository
5. In the front end the systems looks like it's hanged, but an error `Only admins are allowed to create items using external data` occurred at the server side.

## Expected behavior
If the user has the permission to submit content in a particular collection. The system should allow that.

**Comment by tdonohue** [comment]
Possible duplicate of #9114.  I think this is describing that same bug. It's still waiting on a volunteer, as I agree it should be fixed.

**Comment by tdonohue** [comment]
Closing, as this is definitely a duplicate of #9114.  That describes the issue in more detail because currently only Administrators can import from external sources during the submission process.
